feature-settings-impl/src/main/java/io/novafoundation/nova/feature_settings_impl/domain/CloudBackupSettingsInteractor.kt

```diff
@@ -92,12 +91,10 @@ class RealCloudBackupSettingsInteractor(
             .toSortedMap(metaAccountTypeComparator())
     }
 
-    override suspend fun applyBackupAccountDiff(): Result<Unit> {
+    override suspend fun applyBackupAccountDiff(cloudBackupDiff: CloudBackupDiff): Result<Unit> {
         return cloudBackupService.fetchAndDecryptExistingBackupWithSavedPassword()
             .mapCatching { cloudBackup ->
-                val localSnapshot = cloudBackupFacade.fullBackupInfoFromLocalSnapshot()
-                val diff = localSnapshot.localVsCloudDiff(cloudBackup, BackupDiffStrategy.syncWithCloud())
-                cloudBackupFacade.applyBackupDiff(diff, cloudBackup)
+                cloudBackupFacade.applyBackupDiff(cloudBackupDiff, cloudBackup)
```

Comment:
This still suffers from the same issue - you are fetching backup from the cloud and apply to the local state while user agreed only on previous backup we fetched.

The safe flow could be the following:

1. Pass fetched backup to DiffException alongside with cloudBackupDiff
2. Pass the backup to applyBackupAccountDiff (this method)
3. Use passed backup for applyBackupDiff